### PR TITLE
[WIP] draft for lgr signatures

### PR DESCRIPTION
**Pull request by dselivanov**

### CODE REVIEW

**Comment by artemklevtsov** [comment]
Related with #30

File: R/logging.R [hunk 1]
```diff
@@ -32,103 +32,89 @@ Logger = R6::R6Class(
...
-
-#' @rdname logging_constants
-#' @export
-DEBUG = 8L
-
-#' @rdname logging_constants
-#' @export
-TRACE = 9L
+logging_constants = list(
+  'fatal' = 100,
```
Comment (artemklevtsov) [inline R/logging.R]: Why not integers?

Comment (dselivanov) [reply R/logging.R]: lgr use numeric

File: R/RestRserveApplication.R [hunk 2]
```diff
@@ -279,8 +279,8 @@ RestRserveApplication = R6::R6Class(
 
       for (id in mw_ids) {
         mw_name = private$middleware[[id]][["name"]]
-        self$logger$trace(
-          list(request_id = request$request_id,
+        self$logger$trace("",
+          context = list(request_id = request$request_id,
                middleware = mw_name,
                message = sprintf("call %s middleware", mw_flag))
```
[the remark this answers is not in the document]
Comment (dselivanov) [reply R/RestRserveApplication.R]: this way it won't be compatible with lgr - it expects character as msg

[the remark this answers is not in the document]
Comment (artemklevtsov) [reply R/RestRserveApplication.R]: I know. I mean to use `msg` when log entry is simple text and use `context` when nested.

**Comment by artemklevtsov** [comment]
I suggest add test with `lgr` object instead builtin to check compatibility.

**Comment by dselivanov** [comment]
I'm merging now - will address suggestions later

**Pull request merged**